contrib/scripts/devcontainer-setup.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+### enable debug
+
+set -e
+
+### User Setting
+
+KUBERNETES_VERSION=v1.29
+KIND_VERSION=v0.19.0
+CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/main/stable.txt)
+CLI_ARCH=amd64
+if [ "$(uname -m)" = "aarch64" ]; then CLI_ARCH=arm64; fi
+
+### Functions Implementation
+
+function confirmation_prompt() {
+    echo "-------------------------------------------------------------------"
+    echo "This will install kubernetes, kind and cilium-cli packages."
+    echo "*** Basically this needs to be done only once on devcontainer image creation ***"
+    echo "-------------------------------------------------------------------"
+
+    while true; do
+        read -p "Do you want to proceed? (yes/no) " yn
+        case $yn in
+            yes ) echo OK, we will proceed;
+                break;;
+            no ) echo exiting...;
+                exit;;
+            * ) echo invalid response;
+        esac
+    done
+}
+
+function exit_trap() {
+    if [ $? != 0 ]; then
+        echo "Command [$BASH_COMMAND] is failed"
+        exit 1
+    fi
+}
+
+function check_sudo() {
+	if [ "$EUID" -ne 0 ]; then
+		echo "Please run this script with sudo."
+		exit 1
+	fi
+}
```

Comment:
indent is completely broken, i will take a look and fix accordingly.

Reply:
https://github.com/cilium/cilium/pull/31534/commits/c757a55415095dea7942b33670e479a9cb4c539c fixed